src/components/Booklist.js

```diff
@@ -1,24 +1,31 @@
 import { useSelector } from 'react-redux';
 import Book from './Book';
-// import booksSlice from '../redux/books/booksSlice';
 
 function BookList() {
   const {
     booksList,
   } = useSelector((state) => state.books);
 
+  if (!Array.isArray(booksList)) {
+    console.log('booklist is not an array');
+    return null; // or return an appropriate JSX if the data is not available yet
+  }
+
   return (
```

Comment:
- [ ] From the project requirement, as required, we're to use the `useEffect` hook provided by react for fetching external data, and inside of it, we can then dispatch our redux functions. Kindly use the `useEffect` hook as required 👍🏽

<img width="1171" alt="Screenshot 2023-06-24 at 11 49 45 AM" src="https://github.com/sunga12/bookstore/assets/50941074/3c83c175-8fbf-44fd-86bc-1162153824d1">

Reply:
This is done in [App.js](https://github.com/sunga12/bookstore/blob/connect-api/src/App.js).